File: 23.07.2024/Sherlock and Cost.c
#include <stdio.h>
#define max(a, b) ((a) > (b) ? (a) : (b))

int main() {
    int t;
    scanf("%d", &t);
    while (t--) {
        int n;
        scanf("%d", &n);
        int B[n], low = 0, high = 0;
        for (int i = 0; i < n; i++) scanf("%d", &B[i]);
        for (int i = 1; i < n; i++) {
            int low_to_low = low;
            int low_to_high = low + abs(B[i] - 1);
            int high_to_low = high + abs(B[i - 1] - 1);
            int high_to_high = high + abs(B[i] - B[i - 1]);
            low = max(low_to_low, high_to_low);
            high = max(low_to_high, high_to_high);
        }
        printf("%d\n", max(low, high));
    }
    return 0;
}

File: 23.07.2024/Sherlock and Cost.py
def cost(B):
    low, high = 0, 0
    for i in range(1, len(B)):
        low_next = max(low, high + B[i-1] - 1)
        high_next = max(high + abs(B[i] - B[i-1]), low + B[i] - 1)
        low, high = low_next, high_next
    return max(low, high)

B = [10, 1, 10, 1, 10]
print(cost(B))
